src/dune_rules/melange/melange_rules.ml

```diff
@@ -245,6 +245,7 @@ let setup_emit_cmj_rules ~sctx ~dir ~scope ~expander ~dir_contents
         ~dialects:(Dune_project.dialects (Scope.project scope))
         ~modes:`Melange_emit )
   in
+  let* () = Buildable_rules.gen_select_rules sctx compile_info ~dir in
```

Comment:
@rgrinberg I thought this would be enough, but while the rules for the selected modules are added, the targets never end up being produced in the test that I wrote, and the compilation fails with:

```
+  File "main.ml", line 2, characters 16-27:
+  2 | let () = Js.log Foo.message
+                      ^^^^^^^^^^^
+  Error: Unbound module Foo
+  [1]
```